app/models/post.rb

```diff
@@ -1,5 +1,22 @@
 class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
   belongs_to :author, class_name: 'User'
-  has_many :comments, foreign_key: :post_id
-  has_many :likes, foreign_key: :post_id
+
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comments_counter, :integer, default: 0
+  attribute :likes_counter, :integer, default: 0
+
+  after_create :update_posts_counter
+  after_destroy :update_posts_counter
+  after_save :update_post_counter
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
+  def five_most_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
```

Comment:
[OPTIONAL]
- [ ] Nice work using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) to call the `#update_posts_counter` method every time a new post has been saved or destroy :clap: However it would be nice to make the method private since we only need it in the post model.
```suggestion
  def five_most_recent_comments
    comments.order(created_at: :desc).limit(5)
  end
  
  private
  
  def update_posts_counter
    author.update(posts_counter: author.posts.count)
  end
```